.github/workflows/test-go.yml

```
@@ -239,7 +239,9 @@ jobs:
           fi
 
           # On a release branch, add a flag to rerun failed tests
-          if [[ "${{ github.ref_name }}" = "release/*" ]]; then
+          # shellcheck disable=SC2193 # can get false positive for this comparision
+          if [[ -n "${{ github.base_ref }}" && "${{ github.base_ref }}" == release/* ]] || [[ "${{ github.ref_name }}" == release/* ]]
```

I think:

```suggestion
          if [[ "${{ github.base_ref }}" = "release/*" ]] || [[ "${{ github.ref_name }}" = "release/*" ]]
```

should work.